start_admin_sshd.sh

```diff
@@ -8,46 +8,86 @@ mkdir -p /home/ec2-user/.ssh/
 chmod 700 /home/ec2-user/.ssh/
 ssh_host_key_dir="/.bottlerocket/host-containers/admin/etc/ssh"
 ssh_config_dir="/home/ec2-user/.ssh"
+sshd_config="/etc/ssh/sshd_config"
+user_data="/.bottlerocket/host-containers/admin/user-data"
 
-# Populate authorized_keys with all the public keys found in instance meta-data
-# The URLs for keys include an index and the keypair name, e.g.
-# http://169.254.169.254/latest/meta-data/public-keys/0=mykeypair/openssh-key
+set -f
+IFS='
+'
+
+# Is user-data does not contain a JSON block, there's not much point in continuing.
+if ! jq empty ${user_data} &>/dev/null; then
+    echo "${user_data} does not contain valid JSON." >&2
+    exit 1
+fi
+
+# Populate authorized_keys with all the public keys found in api user-data
 ssh_authorized_keys="${ssh_config_dir}/authorized_keys"
 touch ${ssh_authorized_keys}
 chmod 600 ${ssh_authorized_keys}
-public_key_base_url="http://169.254.169.254/latest/meta-data/public-keys/"
-imds_session_token=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 60")
-imds_request_add_header="X-aws-ec2-metadata-token: ${imds_session_token}"
-public_key_indexes=($(curl -H "${imds_request_add_header}" -sf "${public_key_base_url}" \
-    | cut -d= -f1 \
-    | xargs))
 
-for public_key_index in "${public_key_indexes[@]}"; do
-  public_key_data="$(curl -H "${imds_request_add_header}" -sf "${public_key_base_url}/${public_key_index}/openssh-key")"
+userdata_authorized_keys=($(jq -r '.ssh.authorized_keys[]' ${user_data} 2> /dev/null))
+
+for public_key_data in "${userdata_authorized_keys[@]}"; do
   if [[ ! "${public_key_data}" =~ ^"ssh" ]]; then
-    echo "Key ${public_key_data} with index ${public_key_index} looks invalid" >&2
+    echo "Key ${public_key_data} looks invalid" >&2
     continue
   fi
   echo "${public_key_data}" >> "${ssh_authorized_keys}"
   if ! grep -q "${public_key_data}" "${ssh_authorized_keys}"; then
-    echo "Failed to write key with index ${public_key_index} to authorized_keys" >&2
+    echo "Failed to write key ${public_key_data} to ${ssh_authorized_keys}" >&2
     continue
   fi
 done
 
 # If we didn't write any keys at all, there's not much point in continuing
```

Comment:
I don't think we want to exit here as there's a possibility the user is only using CA keys and not SSH keys.